Submodule OpenHD/ohd_interface/lib/wifibroadcast contains modified content
diff --git a/OpenHD/ohd_interface/lib/wifibroadcast/wifibroadcast/src/WBTxRx.cpp b/OpenHD/ohd_interface/lib/wifibroadcast/wifibroadcast/src/WBTxRx.cpp
index ad90955..02c2798 100644
--- a/OpenHD/ohd_interface/lib/wifibroadcast/wifibroadcast/src/WBTxRx.cpp
+++ b/OpenHD/ohd_interface/lib/wifibroadcast/wifibroadcast/src/WBTxRx.cpp
@@ -199,8 +199,19 @@ void WBTxRx::tx_inject_packet(const uint8_t stream_index, const uint8_t* data,
   // occurs)
   assert(data_len + crypto_aead_chacha20poly1305_ABYTES == ciphertext_len);
   // we inject the packet on whatever card has the highest rx rssi right now
-  const bool success =
-      inject_radiotap_packet(m_curr_tx_card.load(), packet_buff, packet_size);
+  bool success;
+  if (m_enable_redundant_tx) {
+    bool any_success = false;
+    for (int i = 0; i < m_pcap_handles.size(); i++) {
+      if (inject_radiotap_packet(i, packet_buff, packet_size)) {
+        any_success = true;
+      }
+    }
+    success = any_success;
+  } else {
+    success =
+        inject_radiotap_packet(m_curr_tx_card.load(), packet_buff, packet_size);
+  }
   if (success) {
     m_tx_stats.n_injected_bytes_excluding_overhead += data_len;
     m_tx_stats.n_injected_bytes_including_overhead += packet_size;
@@ -907,6 +918,10 @@ void WBTxRx::set_passive_mode(bool passive) {
   m_disable_all_transmissions = passive;
 }

+void WBTxRx::set_enable_redundant_tx(bool enable) {
+  m_enable_redundant_tx = enable;
+}
+
 bool WBTxRx::get_card_has_disconnected(int card_idx) {
   if (card_idx >= m_wifi_cards.size()) {
     return true;
diff --git a/OpenHD/ohd_interface/lib/wifibroadcast/wifibroadcast/src/WBTxRx.h b/OpenHD/ohd_interface/lib/wifibroadcast/wifibroadcast/src/WBTxRx.h
index 04ab97b..dd35c85 100644
--- a/OpenHD/ohd_interface/lib/wifibroadcast/wifibroadcast/src/WBTxRx.h
+++ b/OpenHD/ohd_interface/lib/wifibroadcast/wifibroadcast/src/WBTxRx.h
@@ -271,6 +271,8 @@ class WBTxRx {
   // For development only
   std::shared_ptr<DummyLink> get_dummy_link();

+  void set_enable_redundant_tx(bool enable);
+
  public:
   struct SessionExtraData {
     // OpenHD uses different carrier bandwidths for the session key data (20Mhz,
@@ -391,6 +393,8 @@ class WBTxRx {
   AvgCalculator m_packet_decrypt_time;
   AvgCalculator m_tx_inject_time;

+  std::atomic<bool> m_enable_redundant_tx = false;
+
  private:
   // For OpenHD rate control, this method should block until the driver accepted
   // the packet returns true if packet is now in linux kernel / driver hands,
